client/src/pages/UserSettings.tsx

```diff
@@ -47,18 +50,56 @@ const LanguageButtonFin = styled(Button)(() => ({
   border: '2px solid Gray',
   color: 'black',
 }))
+
 const UserSettings = () => {
   // TODO: @finnan - get all the details from db
+  interface User {
+    username: string
+  }
+
+  const [userSub, setUserSub] = useState<string>('')
+  const [userData, setUserData] = useState<UserBySubQuery | null>(null)
+
+  const user = useAuthUser() as User | null
+  
+  useEffect(() => {
+    if (user && user.username) {
+      setUserSub(user.username)
+    }
+  }, [user])
+
+  const fetchedUserData = useGetUserData(userSub)
+
+  useEffect(() => {
+    if(fetchedUserData) {
+      setUserData(fetchedUserData)
+    }
+  }, [fetchedUserData])
```

Comment:
We need to make sure that we are doing this once or only until we get the user data. Passing fetchedUserData as it is might lead to uncontrolled state management. Can we please wrap this with useCallback or some optimizer like useMemo